### PR TITLE
DSPDC-1110 Add support for table fragments

### PR DESCRIPTION
**Pull request by danxmoran**
Still working on comments & cleanup, but the tests show the idea now.

### CODE REVIEW

File: plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/JadeSchemaGenerator.scala [hunk 1]
```diff
@@ -56,13 +68,59 @@ object JadeSchemaGenerator {
...
+  }
+
+  private def flattenFragments(
+    table: MonsterTable,
+    fragments: Map[JadeIdentifier, MonsterTableFragment]
+  ): Either[String, MonsterTable] = {
+    val base: Either[String, MonsterTable] = Right(table.copy(tableFragments = Vector.empty))
+    table.tableFragments.foldLeft(base) { (acc, fragmentId) =>
+      acc.flatMap { baseTable =>
+        fragments.get(fragmentId).toRight(s"No such fragment: $fragmentId").flatMap { subTable =>
```
Comment (danxmoran) [inline plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/JadeSchemaGenerator.scala:100]: I'm going to extract this validation logic into its own method, and wire it up to `sbt test` in the plugin.

File: plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/model/ClassSpec.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.broadinstitute.monster.sbt.model
+
+trait ClassSpec {
```
Comment (danxmoran) [inline plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/model/ClassSpec.scala:3]: Needs doc

File: plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/model/MonsterTableFragment.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.broadinstitute.monster.sbt.model
+
+import io.circe.Decoder
+import io.circe.derivation.{deriveDecoder, renaming}
+
+case class MonsterTableFragment(
```
Comment (danxmoran) [inline plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/model/MonsterTableFragment.scala:6]: Needs doc

**Review by snwessel — APPROVED** [review]
Looks good to me.

**Review by raaidbroad — APPROVED** [review]

**Pull request merged**